### ISSUE TITLE
API consistancy in shuffle default parameter

### ISSUE THREAD

**Issue by MarieSacksick** [probabl-ai/skore#2208]
### What would you like to say?

```python
# %%
import sklearn

X, y = sklearn.datasets.fetch_california_housing(
    return_X_y = True,  as_frame = True
)

# %%
from skore import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y)

# %%
from skore import EstimatorReport
from sklearn.ensemble import HistGradientBoostingRegressor

hist_est_rep = EstimatorReport(
    HistGradientBoostingRegressor(random_state=4),
    X_train=X_train,
    y_train=y_train,
    X_test=X_test,
    y_test=y_test
)

# %%
hist_est_rep.metrics.summarize().frame()

# %%
from skore import CrossValidationReport

hist_cv_rep = CrossValidationReport(
    HistGradientBoostingRegressor(random_state=1),
    X=X,
    y=y
)


# %%
hist_cv_rep.metrics.summarize().frame()
```

It will result in:  

<img width="403" height="558" alt="Image" src="https://github.com/user-attachments/assets/0173648a-e9b9-479e-846f-e0843d36cb2b" /> 

The estimator results are inconsistant with the CV.  

@GaetandeCast idenfied the discrepancy: it's because by default, the parameter shuffle of the TTS is set at True, while the cross validation uses KFold with shuffle set to False.  
I think it creates a situation that is weird and inconsistant between our EstimatorReport & CrossValidationReport. It leads to a weird situation, while we sell that the skore prevents methodological mistakes --> we should align on only one default for shuffling.  

See this other issue https://github.com/probabl-ai/skore/issues/861 which is related.

**Comment by glemaitre** [comment]
> It leads to a weird situation, while we sell that the skore prevents methodological mistakes --> we should align on only one default for shuffling.

The hard part here is choosing the right default and I'm not sure that there is one: it is valid to not shuffle the data apart from when there is structure in the data collection such as consecutive target or hidden group. In the first case, the scores will be really bad while in the second case, it will be over-optimistic. The first case is easy to detect looking at the target. The second one is pretty tough.

Given those, I might advocate for:

- not shuffling the data
- inspect the the target and alert if we detect group of classes or values (not sure how to do for the second one)
- Check the variance of the cross-validated results and check if some scores are bad. If the variance is quite important it could mean that there an underlying structure. We could warn that one might want to use group. Another check is to actually rerun the evaluation with shuffle data and check if the variance disappear. In this case it means that there are groups).

**Comment by auguste-probabl** [comment]
Can still reproduce with

```python
from skore import evaluate
from sklearn.datasets import fetch_california_housing
from sklearn.ensemble import HistGradientBoostingRegressor

X, y = fetch_california_housing(return_X_y=True, as_frame=True)
model = HistGradientBoostingRegressor(random_state=1)
hist_est_rep = evaluate(model, X, y)
print(hist_est_rep.metrics.summarize().frame())
hist_cv_rep = evaluate(model, X=X, y=y, splitter=3)
print(hist_cv_rep.metrics.summarize().frame())

#                   HistGradientBoostingRegressor
# Metric
# R²                                     0.834869
# RMSE                                   0.464029
# MAE                                    0.312884
# MAPE                                   0.179657
# Fit time (s)                           0.332261
# Predict time (s)                       0.006903
#                  HistGradientBoostingRegressor
#                                           mean       std
# Metric
# R²                                    0.687684  0.048511
# RMSE                                  0.642643  0.078688
# MAE                                   0.459319  0.041286
# MAPE                                  0.263502  0.038691
# Fit time (s)                          0.354851  0.045892
# Predict time (s)                      0.008645  0.001922
```
